src/vivarium_public_health/disease/model.py

```diff
@@ -120,12 +97,35 @@ def setup(self, builder: Builder) -> None:
 
         self.configuration_age_start = builder.configuration.population.initialization_age_min
         self.configuration_age_end = builder.configuration.population.initialization_age_max
+
         builder.value.register_value_modifier(
             "cause_specific_mortality_rate",
             self.adjust_cause_specific_mortality_rate,
             requires_columns=["age", "sex"],
         )
-        self.randomness = builder.randomness.get_stream(f"{self.state_column}_initial_states")
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        """Initialize the simulants in the population.
+
+        If all simulants are initialized at age 0, birth prevalence is used.
+        Otherwise, prevalence is used.
+
+        Parameters
+        ----------
+        pop_data
+            The population data object.
+        """
+        if pop_data.user_data.get("age_end", self.configuration_age_end) == 0:
+            initialization_table_name = "birth_prevalence"
+        else:
+            initialization_table_name = "prevalence"
+
+        for state in self.states:
+            state.lookup_tables["initialization_weights"] = state.lookup_tables[
+                initialization_table_name
+            ]
+
+        super().on_initialize_simulants(pop_data)
```

Comment:
Whats the reason for doing this at the end of the method?

Reply:
It seems like it doesn't matter

Reply:
The reason this matters is that `BaseDiseaseState` and its subclasses don't set their initialization weights when they're created. They only create `prevalence` and `birth_prevalence` lookup tables. `Machine` uses the `initialization_weights` lookup table, so we need to set that lookup table appropriately. If `age_end` is 0 we should use birth prevalence and otherwise, prevalence.